pkg/git/factory/gitfactory.go

```diff
@@ -2,43 +2,76 @@ package gitfactory
 
 import (
 	"context"
+	"fmt"
+	"path/filepath"
 
 	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
 	"github.com/aws/eks-anywhere/pkg/git"
+	"github.com/aws/eks-anywhere/pkg/git/gitclient"
 	"github.com/aws/eks-anywhere/pkg/git/gogithub"
 	"github.com/aws/eks-anywhere/pkg/git/providers/github"
 )
 
-type gitProviderFactory struct {
-	GitClient github.GitClient
-}
+type gitProviderFactory struct{}
 
-type Options struct {
-	GithubGitClient github.GitClient
+func New() *gitProviderFactory {
+	return &gitProviderFactory{}
 }
 
-func New(opts Options) *gitProviderFactory {
-	return &gitProviderFactory{GitClient: opts.GithubGitClient}
+func (g *gitProviderFactory) Build(ctx context.Context, cluster *v1alpha1.Cluster, fluxConfig *v1alpha1.FluxConfig, writer filewriter.FileWriter) (git.ProviderClient, git.Client, filewriter.FileWriter, error) {
```

Comment:
it's bit strange to return both gitProvider and gitClient from the `gitProviderFactory` method. Also what's the reason to keep this `gitProviderFactory` struct for if non of its field is used in any method?

Reply:
if using the `git` struct i mentioned above, this method can just return it? to avoid many return values

Reply:
now using `GitTools` for this